### PR TITLE
snaps: do not validate snaps before install/refresh (Fixes LP#1901733)

### PR DESCRIPTION
**Pull request by Saviq**
Branches are not advertised in store channels, the only way to check is to
to check with the store API, which `install` and `refresh` do already.

Fixes https://bugs.launchpad.net/snapcraft/+bug/1901733

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

### CODE REVIEW

File: snapcraft/internal/repo/snaps.py [hunk 1]
```diff
@@ -167,6 +167,8 @@ def is_classic(self) -> bool:
 
     def is_valid(self) -> bool:
         """Check if the snap is valid."""
+        if self.installed and self.get_local_snap_info()["channel"] == self.channel:
```
Comment (sergiusens) [inline snapcraft/internal/repo/snaps.py:170]: the caveat here is that for this to work it has to be installed, which implies destructive-mode, correct?

Comment (Saviq) [reply snapcraft/internal/repo/snaps.py:170]: Or a preexisting environment?

Comment (sergiusens) [reply snapcraft/internal/repo/snaps.py:170]: yes, I was thinking about the bug state after this lands as it alleviates the problem but does not completely fix it, that's all :-)

**Review by sergiusens — APPROVED** [review]

**Review by cmatsuoka — APPROVED** [review]
Thanks Michał, looks good.

**Pull request merged**